sporkfish/statistics.py

```diff
@@ -1,26 +1,131 @@
+from enum import Enum
+from typing import Dict
+
+
+class NodeTypes(Enum):
+    NEGAMAX = "NEGAMAX"
+    NEGAMAX_LAZY_SMP = "NEGAMAX_LAZY_SMP"
+    QUIESCENSE = "QUIESCENSE"
+
+
 class Statistics:
     """A class for tracking statistics related to node visits."""
 
-    def __init__(self, nodes_visited: int):
+    def __init__(self):
         """
         Initialize the Statistics object.
+        Upon initialization, sets the counts of visited nodes, pruned nodes,
+        and nodes stored in the transposition table to zero.
+        """
+        self._nodes_visited = {
+            NodeTypes.NEGAMAX: 0,
+            NodeTypes.NEGAMAX_LAZY_SMP: 0,
+            NodeTypes.QUIESCENSE: 0,
+        }
+        self._pruned = 0
+        self._nodes_in_tt = 0
+
+    def increment_node_visited(
```

Comment:
Reason I did it this way was because I felt TT & pruning are logically very distinct from the node counts. The distinction makes it clear to readers. 

I want to enrich the pruning count to also be a dictionary in the future, categorising different types of pruning. For this reason, I think they deserve different interfaces?

Agree with you the interface can be designed better in general.

Reply:
Im not sure readability is a big concern because when they call the API they need to pass in the enum to increment anyway, like increment(NodeType.QUIESCENCE) or increment(Pruning). I think it mainly becomes an issue when we try to aggregate these, the logic becomes slightly more complicated. Anyway no strong opinions either way